gwpy/timeseries/tests/test_timeseries.py

```diff
@@ -856,6 +856,22 @@ def test_taper(self):
         assert tapered.size == data.size
         utils.assert_allclose(data.value, numpy.cos(10*numpy.pi*t))
 
+        # run the same tests for a user-specified taper duration
+        dtapered = data.taper(duration=0.1)
+        assert dtapered[0].value == 0
+        assert dtapered[-1].value == 0
+        assert dtapered.unit == data.unit
+        assert dtapered.size == data.size
+        utils.assert_allclose(data.value, numpy.cos(10*numpy.pi*t))
+
+        # run the same tests for a user-specified number of samples to taper
+        stapered = data.taper(nsamples=10)
+        assert stapered[0].value == 0
+        assert stapered[-1].value == 0
+        assert stapered.unit == data.unit
+        assert stapered.size == data.size
+        utils.assert_allclose(data.value, numpy.cos(10*numpy.pi*t))
+
```

Comment:
@tjma12, if you decide to keep the use case exception in `TimeSeries.taper()`, could you include a separate unit test for it here? Something like:

```python
        with pytest.raises(ValueError) as exc:
            data.taper(duration=0.1, nsamples=10)
        assert str(exc.value) == ("only one of 'duration' or 'nsamples' "
                                  "may be provided as arguments")
```

Reply:
Removed this check in new commit